gdmlGeneratorV1_materials.pl

```diff
@@ -9,224 +9,281 @@
-		{
-		}
-		elsif (index($opt_L, "5") >= 0 && index($opt_L, "trans") >= 0 && substr(trim($fields[0]), 0, 5) == "51005")
-		{	
-		}
-		elsif (index($opt_L, "5") >= 0 && index($opt_L, "closed") >= 0 && substr(trim($fields[0]), 0, 5) == "52002")
-		{
-		}
-		elsif (index($opt_L, substr(trim($fields[0]), 0, 1 )) >= 0 && substr(trim($fields[0]), 0, 1) != "5" &&
-            ((index($opt_L, "trans") >= 0 && substr(trim($fields[0]), 3, 2) == "01") ||
-            ((index($opt_L, "open") >= 0 || (index($opt_L, "closed") < 0 && index($opt_L, "trans") < 0)) && substr(trim($fields[0]), 3, 2) == "03") || 
-            (index($opt_L, "closed") >= 0 && substr(trim($fields[0]), 3, 2) == "2"))) 
-		{	
-		}
-		else
-		{
-			next;
-		}
-	}
-
-	$index[$i]=trim($fields[0]);            # Get rid of initial and trailing white spaces.
-	$x[$i]=trim($fields[1]);
-	$y[$i]=trim($fields[2]);
-	$z[$i]=trim($fields[3]);
-	$dx[$i]=trim($fields[4]);
-	$dy[$i]=trim($fields[5]);
-	$dz[$i]=trim($fields[6]);
-	$rx[$i]=trim($fields[7]);
-	$ry[$i]=trim($fields[8]);
-	$rz[$i]=trim($fields[9]);
-	$quartzCutAngle[$i]=trim($fields[10]);
-	$refTopOpeningAngle[$i]=trim($fields[11]);
-	$dzRef[$i]=trim($fields[12]);
-	$dxLg[$i]=trim($fields[13]);
-	$dyLg[$i]=trim($fields[14]);
-	$dzLg[$i]=trim($fields[15]);
-	$lgTiltAngle[$i]=trim($fields[16]);
-	$dxPmt[$i]=trim($fields[17]); 		
-	$dyPmt[$i]= trim($fields[18]);
-	$dzPmt[$i]= trim($fields[19]);
-	$drPmt[$i]= trim($fields[20]);
-	$dtWall[$i]= trim($fields[21]); 
-	$dtReflector[$i]= trim($fields[22]); 
-    $dzLgExtra[$i]=(0.55*$dxPmt[$i])*sin($lgTiltAngle[$i]+$ry[$i]);
-	$i=$i+1;
-}
+open( $data, '<', $opt_D );      # Open csv file.
+$i = 0;
+while ( $line = <$data> ) {      # Read each line till the end of the file.
+    if ( $line =~ /^\s*$/ ) {    # Check for empty lines.
+        print
+          "String contains 0 or more white-space character and nothing else.\n";
+    }
+    else {
+        chomp $line;
+        @fields = split( ",", $line );    # Split the line into fields.
+        if ( $opt_L != "" ) {
+            if (
+                index( $opt_L, "5" ) >= 0
+                && (
+                    index( $opt_L, "open" ) >= 0
+                    || (   index( $opt_L, "closed" ) < 0
+                        && index( $opt_L, "trans" ) < 0 )
+                )
+                && substr( trim( $fields[0] ), 0, 5 ) == "50001"
+              )
+            {
+            }
+            elsif (index( $opt_L, "5" ) >= 0
+                && index( $opt_L, "trans" ) >= 0
+                && substr( trim( $fields[0] ), 0, 5 ) == "51005" )
+            {
+            }
+            elsif (index( $opt_L, "5" ) >= 0
+                && index( $opt_L, "closed" ) >= 0
+                && substr( trim( $fields[0] ), 0, 5 ) == "52002" )
+            {
+            }
+            elsif (
```

Comment:
I think this piece of logic was trying to include all 3 sectors if the words open, trans or closed weren't included at all.